File: forbiddeninteger.py
def solve():
    #function
    n , k , x = map(int,input().split())
    if(x!=1):
        print("YES")
        print(n)
        for i in range(n):
            print(1,end=" ")
        print()
    else:
        if(k==1):
            print("NO")
        elif(k==2):
            if(n%2==0):
                print("YES")
                print(n//2)
                for i in range(n//2):
                    print(2,end=" ")
                print()
            else:
                print("NO")
        else:
            if(n%2==0):
                print("YES")
                print(n//2)
                for i in range(n//2):
                    print(2,end=" ")
                print()
            else:    
                a = n//2
                b = (a-1)*2   
                print("YES")
                print(a)
                for i in range(a-1):
                    print(2,end=" ")
                print(3)
                print()                         


for _ in range(int(input())):
    solve()

File: forbiddeninteger.cpp
#include<bits/stdc++.h>
using namespace std;

void solve()
{
    //Function
    int n,k,x;
    cin>>n>>k>>x;
    int a,b;
    if(x!=1){
        cout<<"YES"<<endl;
        cout<<n<<endl;
        for(int i=1;i<=n;i++){
            cout<<1<<" ";
        }
        cout<<endl;
    }
  
    else{
        for(int i=2;i<=k;i++){
            a = n%i;
            if((a>1 && a<=k)){
                cout<<"YES"<<endl;
                cout<<n/i + 1<<endl;
                b = i;
                for(int i=1;i<=n/b;i++){
                    cout<<b<<" ";
                }
                cout<<a<<endl;  
                goto end;
            }
            if(a==0){
                cout<<"YES"<<endl;
                cout<<n/i<<endl;
                b = i;
                for(int i=1;i<=n/b;i++){
                    cout<<b<<" ";
                }  
                cout<<endl;
                goto end;
            }
        }
        cout<<"NO"<<endl;
        end:;
        
    }
}

int main()
{
    //main function
    int t;
    cin>>t;
    while(t--){
        solve();
    }
    
    return 0;
}